### PR TITLE
Add withAllowedLateness with Closing Behavior to Window

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

This makes the static constructors for withAllowedLateness symmetric to
the PTransform builder methods. It also allows references to
Window#withAllowedLateness(Duration, ClosingBehavior).

### CODE REVIEW

**Comment by tgroh** [comment]
R: @dhalperi

**Comment by dhalperi** [comment]
R: -@dhalperi @kennknowles

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java [hunk 1]
```diff
@@ -224,6 +224,22 @@
   }
 
   /**
+   * Override the amount of lateness allowed for data elements in the pipeline. Like
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java]: elements in the output PCollection, and downstream until it is explicitly set again

**Comment by kennknowles** [comment]
In the commit message, you mean static constructors of `WindowingStrategy`?

**Comment by tgroh** [comment]
No, with the methods on `Window.Bound` that modify that `PTransform`

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java (hunk 1 above)
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java]: Done.

**Comment by kennknowles** [comment]
Ah, got it. This is no prob at all. LGTM.

**Pull request closed without merging**